internal/graph/graph.go

```diff
@@ -686,6 +686,8 @@ func defaultLabelFormat(v int64, key string) string {
 func (m TagMap) findOrAddTag(label, unit string, value int64) *Tag {
 	l := m[label]
 	if l == nil {
+		// Filter out all null characters since dot is not able to parse them
+		label = strings.ReplaceAll(label, "\000", "")
```

Comment:
Is it possible to add a test case?

It would also be good to mention why this can happen. Also, are null characters the only ones that need to be filtered out?

Reply:
I also think escaping rather than filtering out these characters would be a more correct approach.